packages/components/src/withHorizontalScroll/withHorizontalScroll.ts

```diff
@@ -77,12 +81,34 @@ const scrollRightToStep = (
   );
 };
 
+const calculateActualScrollForIndex = (
+  itemRefs: HTMLElement[],
+  scrollStep: number,
+  toLeft: boolean
+) => {
+  if (toLeft) {
+    const firstVisibleIndex = findFirstVisibleIndex(itemRefs);
+    const actualScrollForIndex =
+      firstVisibleIndex < scrollStep ? 0 : firstVisibleIndex - scrollStep;
+    return actualScrollForIndex;
+  } else {
```

Comment:
we can don't use else there as we have return in if block. Code will be a little bit clear.